crates/reth-rbuilder/src/main.rs

```diff
@@ -83,7 +84,12 @@ fn main() {
                         .with_components(EthereumNode::components())
                         .with_add_ons(EthereumAddOns::default())
                         .on_rpc_started(move |ctx, _| {
-                            spawn_rbuilder(ctx.provider().clone(), extra_args.rbuilder_config);
+                            let ipc_path = if ctx.config().rpc.is_ipc_enabled() {
+                                Some(PathBuf::from(ctx.config().rpc.ipc_path()))
+                            } else {
+                                None
+                            };
```

Comment:
Can you move this insde spawn_rbuilder or as a func to avoid more copy/paste?

Reply:
Moved it to draft, it did not show the improvement I was expecting.